pkg/services/datasources/service/datasource.go

```diff
@@ -528,6 +528,11 @@ func (s *Service) UpdateDataSource(ctx context.Context, cmd *datasources.UpdateD
 			}
 		}
 
+		// TODO: we will eventually remove this check for moving the resource to it's separate API
+		if s.features != nil && s.features.IsEnabled(ctx, featuremgmt.FlagTeamHttpHeaders) && !cmd.OnlyUpdateLBACRulesFromAPI && datasources.CheckTeamHTTPHeadersDiff(dataSource.JsonData, cmd.JsonData) {
```

Comment:
Alternative approach, probably safer security wise:
```suggestion
		if s.features != nil && s.features.IsEnabled(ctx, featuremgmt.FlagTeamHttpHeaders) && !cmd.OnlyUpdateLBACRulesFromAPI {
		  s.logger.Debug("Override LBAC rules with stored ones.")
		  datasources.OverrideTeamHTTPHeaders(dataSource.JsonData, cmd.JsonData)
```

Reply:
this solution works amazing. We do not check for diffs, nor do we have a "error" that we introduce. Just update the previousrules if they exists.

updated to account for `nil`
```go
		// TODO: we will eventually remove this check for moving the resource to it's separate API
		if s.features != nil && s.features.IsEnabled(ctx, featuremgmt.FlagTeamHttpHeaders) && !cmd.OnlyUpdateLBACRulesFromAPI {
			s.logger.Warn("Overriding LBAC rules with stored ones. Use updateDatasourceLBACRules API to update team HTTP headers instead.")
			previousRules := dataSource.JsonData.Get("teamHttpHeaders")
			if previousRules != nil {
				cmd.JsonData.Set("teamHttpHeaders", previousRules)
			}
		}
```